xapian-core/include/xapian/mset.h

```diff
@@ -74,6 +77,30 @@ class XAPIAN_VISIBILITY_DEFAULT MSet {
     /// Destructor.
     ~MSet();
 
+    /** Assigns new weights to items
+     * This functon is used in letor to update the weights assigned by the ranker
+     * It updates weights of all the elements in items as well as updating the max_attained
```

Comment:
That would be more readable as a comment if you put a line of 3 backticks ( ```) before and after as that renders as a preformatted block - I did an edit to do that.

The `Iterator ...` paragraph seems a bit woolly - weights don't get dereferenced (they're returned by the iterator when dereferenced), the "original order of the documents" could be taken to mean "docid order" rather than the order in the `MSet`, and it talks about the order twice which might make the reader think it's talking about two different things.

I'd suggest `The weights returned by the iterator are assigned to elements of the MSet in rank order.`

You've also indented the continuation of the `@exception` line too much - `elements` and `documents` should line up under `Xapian`.